sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/cardinality/ApproximateDistinct.java

```diff
@@ -0,0 +1,352 @@
+          mergedAccum.addAll(accum);
+        } catch (CardinalityMergeException e) {
+          // Should never happen because only HyperLogLogPlus accumulators are instantiated.
+          LOG.error("The accumulators cannot be merged : " + e.getMessage(), e);
+        }
+      }
+      return mergedAccum;
+    }
+
+    @Override
+    public Coder<HyperLogLogPlus> getAccumulatorCoder(CoderRegistry registry, Coder inputCoder) {
+      return HyperLogLogPlusCoder.of();
+    }
+
+    @Override
+    public void populateDisplayData(DisplayData.Builder builder) {
+        super.populateDisplayData(builder);
+        builder
+                .add(DisplayData.item("p", p)
+                        .withLabel("Precision"))
+                .add(DisplayData.item("sp", sp)
+                        .withLabel("Sparse Representation precision"));
+      }
+  }
+
+  static class HyperLogLogPlusCoder extends CustomCoder<HyperLogLogPlus> {
+
+    private static final HyperLogLogPlusCoder INSTANCE = new HyperLogLogPlusCoder();
+
+    private static final ByteArrayCoder BYTE_ARRAY_CODER = ByteArrayCoder.of();
+
+    public static HyperLogLogPlusCoder of() {
+      return INSTANCE;
+    }
+
+    @Override public void encode(HyperLogLogPlus value, OutputStream outStream) throws IOException {
+      if (value == null) {
+        throw new CoderException("cannot encode a null HyperLogLogPlus sketch");
+      }
+      BYTE_ARRAY_CODER.encode(value.getBytes(), outStream);
+    }
+
+    @Override public HyperLogLogPlus decode(InputStream inStream) throws IOException {
+      return HyperLogLogPlus.Builder.build(BYTE_ARRAY_CODER.decode(inStream));
+    }
+
+    @Override public boolean isRegisterByteSizeObserverCheap(HyperLogLogPlus value) {
+      return true;
+    }
+
+    @Override protected long getEncodedElementByteSize(HyperLogLogPlus value) throws IOException {
+      if (value == null) {
+          throw new CoderException("cannot encode a null HyperLogLogPlus sketch");
+      }
+      return value.sizeof();
+    }
+  }
+
+    /**
+     * Computes the precision based on the desired relative error.
+     *
+     * <p>According to the paper, the mean squared error is bounded by the following formula :
+     * <pre>b(m) / sqrt(m)
+     * Where m is the number of buckets used (p = log2(m) )
+     * and b(m) < 1.106 for m > 16 ( p > 4).
+     * </pre>
+     *
+     * <br><b>WARNING : </b>
+     * <br>This does not means relative error in the estimation <b>can't</b> be higher.
+     * <br>This only means that, on average, the relative error will be
+     * lower than the desired relative error.
+     * <br>Nevertheless, the more elements arrive in the stream, the lower the variation will be.
+     * <br>Indeed, this is like when you throw a dice thousands or millions of time :
+     * The relative frequency of each different result {1,2,3,4,5,6} will get closer to 1/6.
+     *
+     * @param relativeError the mean squared error should be in the interval ]0;1]
```

Comment:
Yeah oops I put a ';' instead of a comma. I meant the range from 0 to 1 excluding 0. 
Normally ]0,1] should be ok isn't it ?